Init check in for Android AICamera  with CMake and paddle lib

nickyfantasy opened the pull request.

nickyfantasy committed:
Init check in for Android AICamera  with CMake and paddle lib

nickyfantasy committed:
Init check in for Android AICamera  with CMake and paddle lib

Xreki commented:
Please don't add the libraries of Paddle to the project. And link the static libraries of Paddle, you can use [FindPaddle.cmake](https://github.com/Xreki/PaddleNativeDemo/blob/master/android/optical_character_recognization/FindPaddle.cmake) and write a similar [CMakeList.txt](https://github.com/Xreki/PaddleNativeDemo/blob/master/android/optical_character_recognization/CMakeLists.txt). This [paddle_jni.map](https://github.com/Xreki/PaddleNativeDemo/blob/master/android/optical_character_recognization/paddle_jni.map) can also be used in this demo to reduce the generated `.so` 's size.

Xreki committed:
Add a ssd demo running on linux (#29)

* Add configurarion of ssd model and main source files.

* Add pre-commit configs.

* Support the use of OpenCV to read image.

* Create README.md

* Define class ImageRecognizer to simplify the inferfaces.

* Print more result's information.

* Reorganize the sub-directory of demo.

* Use threshold to profile the result.

* Add script file to download model.

* Add function resize_hwc to resize input image, and move the substract of means into ImageRecognizer::infer().

* Refine the code format.

* Enable the input image to rotate 0, 90, 180 and 270.

* Fix out of range crash and set memory pool size of Paddle to 0.

* Enable the transferring from BGR to RGB.

* Draw rectangles to show the results.

* Rename demo to Demo.

* Reorganize the directory.

* Reorganize the image related code.

NHZlX committed:
add mobilenet ssd pascal model

NHZlX committed:
modify format

NHZlX committed:
refine inference demo doc (#37)

* refine inference demo doc

* Punctuation modification

* format refine

* the demo link the static lib, not the dynamic one

nickyfantasy committed:
Init Commit for iOS AICamera Demo (#26)

*  Init Commit for iOS AICamera Demo
    -Integrate with Paddle libs
    -Integrate with C++ code from liuyiqun
    -Create Objective C wrapper and bridging header to call from client in Swift
    -Able to capture video from device camera and output image buffer
    -Create hardcode data to draw bounding box and label

* fall back to swift 3 and finish objective C wrapper

* Integrate with latest C++ code with resize and pass in byte array

* successfully get results from infer() and draw bounding box

* Integrate with rotate logic, finalized and clean up code

* Add front camera and new model, add settings for different models, accuracy threshold and time refresh rate

* Fix issue where default is front camera

* fix iphone X layout issue

* format settings accuracy label

* change app name to PDCamera
Add paddle logo

* Add README, remove 2 models and update settings, rename to PDCamera

* clean up code, add image_utils.cpp, add screenshot and QR code, update README

Xreki committed:
Correct the link of images in README of iOS demo AICamera and add it to the main README. (#44)

Xreki committed:
Fix linking problem of protobuf-3.2. (#45)

yeyupiaoling committed:
add run path (#47)

NHZlX committed:
Merge pull request #40 from NHZlX/mobilenet_ssd_pascal_model

add mobilenet ssd pascal model

hedaoyuan committed:
Move build_for_minimum_size.

hedaoyuan committed:
Move tools into deployment.

hedaoyuan committed:
Move configuration files.

Commit:
Merge pull request #53 from hedaoyuan/develop

Refine the directory structure

hedaoyuan committed:
Merge branch 'develop' of github.com:PaddlePaddle/Mobile into refine

hedaoyuan committed:
Modify the directory name.

Commit:
Merge pull request #55 from hedaoyuan/refine

Refine `models` directory

nickyfantasy committed:
Able to load paddle library, check in FindPaddle.cmake and update CMakeLists.txt

Xreki committed:
Update the README of how to benchmark Paddle on Android and add the Chinese version (#49)

* Add Chinese README of how to benchmark Paddle on Android.

* Add the output of inference demo.

* Fix typo and add the link of flowers102.

nickyfantasy committed:
Remove paddle lib from git

nickyfantasy committed:
clean up
